src/__tests__/ariaAttributes.js

```diff
@@ -0,0 +1,94 @@
+import {render} from './helpers/test-utils'
+
+test('`selected` throws on unsupported roles', () => {
+  const {getByRole} = render(`<input aria-selected="true" type="text">`)
+  expect(() =>
+    getByRole('textbox', {selected: true}),
+  ).toThrowErrorMatchingInlineSnapshot(
+    `"\\"aria-selected\\" is not supported on role \\"textbox\\"."`,
+  )
+})
+
+test('`selected: true` matches `aria-selected="true"` on supported roles', () => {
+  const {getAllByRole} = render(`
+<select>
+  <option selected id="selected-native-option" />
+  <option id="unselected-native-option" />
+</select>
+<div role="listbox">
+  <div role="option" aria-selected="true" id="selected-listbox-option" />
+  <div role="option" aria-selected="false" id="unselected-listbox-option" />
+  <div role="option" id="not-selectable-listbox-option" />
+</div>
+<div role="tree">
+  <div role="treeitem" aria-selected="true" id="selected-treeitem" />
+  <div role="treeitem" aria-selected="false" id="unselected-treeitem" />
+  <div role="treeitem" id="not-selectable-treeitem" />
+</div>
+<input type="radio" aria-selected="true" id="selected-native-radio" />
+<input type="radio" aria-selected="false" id="unselected-native-radio" />
```

Comment:
Is this not implied for radio buttons? Do people have to explicitly add the attribute for everything?

Reply:
I had this originally but `aria-checked` is different to `aria-selected`. `aria-checked` is implied.

To be honest I don't really understand why you'd want aria-selected on a radio. Apparently assistive technology "may" imply it from focus but that wouldn't apply here.

Right now it's only there for parity with aria-query. 

Though it is removed from the latest [aria editors draft](https://rawgit.com/w3c/aria/master/#aria-selected) and from what I've been told from jsdom folks (who work at google) implementors only use editors drafts.

I'll open an issue on `aria-query` and see how they feel about following editors draft instead. Though response time is quite low so this might take a while.

Reply:
It's also on the 1.2 working draft. Probably a good time to send a PR to aria-query with aria 1.2

Reply:
Let's remove this test since it's no longer part of aria 1.2 anyway.